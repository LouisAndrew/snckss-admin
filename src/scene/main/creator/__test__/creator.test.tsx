// how to test component with node env?

import React, { Suspense } from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { render, cleanup, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'

import Creator from '..'
import { FirebaseAppProvider } from 'reactfire'
import config from '../../../../lib/firebase-config'

describe('Creator element inside main scene', () => {
        beforeEach(() => {
                jest.resetModules()
        })
        afterEach(cleanup)
        const el = (
                <FirebaseAppProvider firebaseConfig={config}>
                        <Suspense fallback={<h1></h1>}>
                                <Creator />
                        </Suspense>
                </FirebaseAppProvider>
        )

        it('env variables matches', () => {
                expect(process.env.REACT_APP_API_KEY).toBeDefined()
        })

        it('renders without crashing', () => {
                const div = document.createElement('div')
                ReactDOM.render(el, div)
        })

        // TODO: test behavior on state change
        // it('renders correctly', () => {

        //         const { getByTestId } = render()
        // })

        // TODO: how to render when state changes.
        // it('renders editor component when link is clicked', () => {
        //         const { getByText, getByTestId } = render(el)

        //         const categoriesLink: HTMLElement = getByText('Categories')
        //         fireEvent.click(categoriesLink)

        //         const editorEl: HTMLElement = getByTestId('editor')
        //         expect(editorEl).toBeInTheDocument()
        // })

        it('matches snapshot', () => {
                const tree = renderer.create(el).toJSON()
                expect(tree).toMatchSnapshot()
        })
})
