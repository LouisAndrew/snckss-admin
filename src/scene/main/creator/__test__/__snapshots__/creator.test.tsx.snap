// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creator element inside main scene matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-lg-4 col-xl-4"
      m="4"
    >
      <h3>
        Select what you want to create
      </h3>
      <ul
        className="nav flex-column nav-tabs"
      >
        <li
          className="categories nav-item"
        >
          <a
            className="nav-link"
            onClick={[Function]}
          >
            Categories
          </a>
        </li>
        <li
          className="brands nav-item"
        >
          <a
            className="nav-link"
            onClick={[Function]}
          >
            Brands
          </a>
        </li>
        <li
          className="products nav-item"
        >
          <a
            className="nav-link"
            onClick={[Function]}
          >
            Products
          </a>
        </li>
      </ul>
    </div>
    <div
      className="col-lg-8 col-xl-8"
      m="8"
    >
      <div
        className="products-creator creator"
      >
        <div
          className="back icon"
          onClick={[Function]}
        >
          <svg
            dangerouslySetInnerHTML={
              Object {
                "__html": "<path d=\\"M20 11v2H8l5.5 5.5l-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5L8 11h12z\\" fill=\\"currentColor\\"/>",
              }
            }
            focusable={false}
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "transform": "rotate(360deg)",
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          />
        </div>
        <h1>
          Products
        </h1>
        <div
          className="form-group"
        >
          <label
            className=""
            htmlFor="name"
          >
            Product's name
          </label>
          <input
            className="form-control"
            id="name"
            onChange={[Function]}
            placeholder="Add product's name here!"
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <label
            className=""
            htmlFor="desc"
          >
            Product's description
          </label>
          <textarea
            className="form-control"
            id="desc"
            onChange={[Function]}
            placeholder="Add product's description here!"
            value=""
          />
        </div>
        <div
          className="available form-group"
        >
          <label
            className=""
            htmlFor="available"
          >
            Is this product available
          </label>
          <input
            className="form-check-input"
            defaultChecked={false}
            id="available"
            name="available"
            onChange={[Function]}
            type="checkbox"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            className=""
            htmlFor="date"
          >
            Product arriving date
          </label>
          <input
            className="form-control"
            defaultValue="2020-07-17"
            id="date"
            onChange={[Function]}
            type="date"
          />
        </div>
        <div
          className="adder"
        >
          <label
            className=""
          >
            Product images
          </label>
          <button
            aria-label={null}
            className="btn btn-dark"
            onClick={[Function]}
            type="button"
          >
            <svg
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<path d=\\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\\" fill=\\"currentColor\\"/>",
                }
              }
              focusable={false}
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style={
                Object {
                  "transform": "rotate(360deg)",
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            />
          </button>
        </div>
        <div
          className="form-group"
        >
          <label
            className=""
            htmlFor="select"
          >
            Add a product variation
          </label>
          <select
            className="form-control"
            id="select"
            name="select"
            onChange={[Function]}
            value=""
          >
            <option
              value="default"
            >
              Select multiple
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
</div>
`;
